# 개수만 확인

In [1]:
from glob import glob
import os
from tqdm import tqdm

In [7]:
txt_path = r'C:\Users\user\parsing\result\train'
test_list = glob(os.path.join(txt_path,'*.txt'))
tot_count = 0
count = [0] * 20
empty_cnt = 0

for file in tqdm(test_list):
    with open(file, 'r') as f:
        lines = f.readlines()
    if not lines:
        empty_cnt += 1
    else:
        for line in lines:
            tot_count += 1
            index = int(line.split()[0])
            # if index == 7:
            #     print(file)    
            count[index] += 1

print('총 labeling 개수 : ', tot_count)

print(count)
print('셔츠 : ', count[0])
print('맨투맨 : ', count[1])
print('자켓 : ', count[2])
print('조끼 : ', count[3])
print('바지 : ', count[4])
print('반바지 : ', count[5])
print('치마 : ', count[6])
print('코트 : ', count[7])
print('드레스 : ', count[8])
# print('검은색 상의 : ', count[15])
# print('검은색 외투 : ', count[16])

100%|██████████| 826328/826328 [00:30<00:00, 27224.47it/s]

총 labeling 개수 :  1232094
[166621, 336545, 84296, 13166, 289767, 52553, 105647, 24812, 158687, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
셔츠 :  166621
맨투맨 :  336545
자켓 :  84296
조끼 :  13166
바지 :  289767
반바지 :  52553
치마 :  105647
코트 :  24812
드레스 :  158687


### Training set

100%|██████████| 826328/826328 [00:30<00:00, 27224.47it/s]<br>
총 labeling 개수 :  1232094<br>
[166621, 336545, 84296, 13166, 289767, 52553, 105647, 24812, 158687, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]<br>
셔츠 :  166621<br>
맨투맨 :  336545<br>
자켓 :  84296<br>
조끼 :  13166<br>
바지 :  289767<br>
반바지 :  52553<br>
치마 :  105647<br>
코트 :  24812<br>
드레스 :  158687

___

### Validation set

100%|██████████| 103200/103200 [00:03<00:00, 28877.06it/s]<br>
총 labeling 개수 :  153766<br>
[20798, 41924, 10565, 1661, 36160, 6531, 13211, 3117, 19799, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]<br>
셔츠 :  20798<br>
맨투맨 :  41924<br>
자켓 :  10565<br>
조끼 :  1661<br>
바지 :  36160<br>
반바지 :  6531<br>
치마 :  13211<br>
코트 :  3117<br>
드레스 :  19799

In [3]:
print(len(os.listdir(r'C:\Users\user\parsing\result\train')))
print(len(os.listdir(r'C:\Users\user\parsing\result\val')))

826328
103200


## 빈파일 확인

In [159]:
from glob import glob
import os
from tqdm import tqdm

In [5]:
path = r'C:\Users\user\parsing\result\train'
txt_list = glob(os.path.join(path, '*.txt'))
empty_cnt = 0

for file in tqdm(txt_list):
    with open(file, 'r') as f:
        lines = f.readlines()
    if not lines:
        empty_cnt += 1
print('빈 labeling 개수 : ', empty_cnt)

100%|██████████| 826328/826328 [32:20<00:00, 425.91it/s]

빈 labeling 개수 :  0


## 삭제

In [ ]:
# del_path = txt_path#r'C:\Users\user\parsing\result\train'

# txt1_list = glob(os.path.join(path, '*.txt'))
# with open(r'C:\Users\user\parsing\train_delfile.txt','r') as f:
#     del_list = f.readlines()
# print(del_list[:5])
# for file in tqdm(del_list):
#     des_file = os.path.join(r'destination', file)
#     # os.remove(des_file)
    
# txt2_list = glob(os.path.join(path, '*.txt'))
# for file in tqdm(txt_list):
#     with open(file, 'r') as f:
#         lines = f.readlines()
#     if not lines:
#         des_file = os.path.join(r'destination', file)
#         # os.remove(des_file)
